pkg/ebpf/events_definitions.go

```diff
@@ -5428,6 +5442,15 @@ var EventsDefinitions = map[int32]EventDefinition{
 	DnsRequest: {
 		ID32Bit: sys32undefined,
 		Name:    "dns_request",
+		Probes: []probeDependency{
```

Comment:
I actually think here we can keep the probes empty, same we do with container_create, since this is a derived event from NetPacket, it should declare the NetPacket event as a dependency.

Reply:
Example with dns_request only:

```
$ sudo ./dist/tracee-ebpf -o format:json -o option:parse-arguments -o option:detect-syscall -trace comm=bash -trace follow -trace event=dns_request -trace net=docker0 -capture net=docker0

{"timestamp":1655230060646770986,"processorId":0,"processId":13,"cgroupId":0,"threadId":13,"parentProcessId":9,"hostProcessId":3782544,"hostThreadId":3782544,"hostParentProcessId":3782540,"userId":0,"mountNamespace":4026533391,"pidNamespace":4026533430,"processName":"http","hostName":"","containerId":"c1ab59ed5848dee2c17cf334cf6c6a4bbfea73a9ead52bf58ef5f3f394a35501","containerImage":"","containerName":"","podName":"","podNamespace":"","podUID":"","eventId":"4001","eventName":"dns_request","argsNum":2,"returnValue":0,"stackAddresses":null,"args":[{"name":"metadata","type":"trace.PktMeta","value":{"src_ip":"172.17.0.2","dst_ip":"1.1.1.1","src_port":33486,"dst_port":53,"protocol":17,"packet_len":78,"iface":"docker0"}},{"name":"dns_questions","type":"[]bufferdecoder.DnsQueryData","value":[{"query":"archive.ubuntu.com","queryType":"A","queryClass":"IN"}]}]}

{"timestamp":1655230060647083330,"processorId":0,"processId":12,"cgroupId":0,"threadId":12,"parentProcessId":9,"hostProcessId":3782543,"hostThreadId":3782543,"hostParentProcessId":3782540,"userId":0,"mountNamespace":4026533391,"pidNamespace":4026533430,"processName":"http","hostName":"","containerId":"c1ab59ed5848dee2c17cf334cf6c6a4bbfea73a9ead52bf58ef5f3f394a35501","containerImage":"","containerName":"","podName":"","podNamespace":"","podUID":"","eventId":"4001","eventName":"dns_request","argsNum":2,"returnValue":0,"stackAddresses":null,"args":[{"name":"metadata","type":"trace.PktMeta","value":{"src_ip":"172.17.0.2","dst_ip":"1.1.1.1","src_port":34317,"dst_port":53,"protocol":17,"packet_len":79,"iface":"docker0"}},{"name":"dns_questions","type":"[]bufferdecoder.DnsQueryData","value":[{"query":"security.ubuntu.com","queryType":"A","queryClass":"IN"}]}]}
```

and with both:

```
$ sudo ./dist/tracee-ebpf -o format:json -o option:parse-arguments -o option:detect-syscall -trace comm=bash -trace follow -trace event=dns_request -trace event=net_packet -trace net=docker0 -capture net=docker0

{"timestamp":1655230109736500168,"processorId":0,"processId":215,"cgroupId":0,"threadId":215,"parentProcessId":211,"hostProcessId":3795107,"hostThreadId":3795107,"hostParentProcessId":3795103,"userId":0,"mountNamespace":4026533391,"pidNamespace":4026533430,"processName":"http","hostName":"","containerId":"c1ab59ed5848dee2c17cf334cf6c6a4bbfea73a9ead52bf58ef5f3f394a35501","containerImage":"","containerName":"","podName":"","podNamespace":"","podUID":"","eventId":"4000","eventName":"net_packet","argsNum":1,"returnValue":0,"stackAddresses":null,"args":[{"name":"metadata","type":"trace.PktMeta","value":{"src_ip":"172.17.0.2","dst_ip":"1.1.1.1","src_port":33780,"dst_port":53,"protocol":17,"packet_len":78,"iface":"docker0"}}]}

{"timestamp":1655230109736500168,"processorId":0,"processId":215,"cgroupId":0,"threadId":215,"parentProcessId":211,"hostProcessId":3795107,"hostThreadId":3795107,"hostParentProcessId":3795103,"userId":0,"mountNamespace":4026533391,"pidNamespace":4026533430,"processName":"http","hostName":"","containerId":"c1ab59ed5848dee2c17cf334cf6c6a4bbfea73a9ead52bf58ef5f3f394a35501","containerImage":"","containerName":"","podName":"","podNamespace":"","podUID":"","eventId":"4001","eventName":"dns_request","argsNum":2,"returnValue":0,"stackAddresses":null,"args":[{"name":"metadata","type":"trace.PktMeta","value":{"src_ip":"172.17.0.2","dst_ip":"1.1.1.1","src_port":33780,"dst_port":53,"protocol":17,"packet_len":78,"iface":"docker0"}},{"name":"dns_questions","type":"[]bufferdecoder.DnsQueryData","value":[{"query":"archive.ubuntu.com","queryType":"A","queryClass":"IN"}]}]}
```

Reply:
You're right I misread the code. 
It does make me confused about the net_packet event now, since it seems the ebpf code can generate it, but it can also be derived inside the ebpf code into a dns_request for example. But then why would we derive it back? Obviously not a question for the scope of this PR though.

Reply:
Yep, it was done like that so the DNS event would also be capable of generating a net_event that generated it. Other way of approaching it would be to depend on net_event and to generate a DNS event from it (and others depending on the protocol handlers). Problem with that would be that every network event would depend on net_event (which has many probes). Let's talk about it, definitely.